File: app/src/main/java/jp/chau2chaun2/kotlindatabindingsample/di/AppModule.java
package jp.chau2chaun2.kotlindatabindingsample.di;

import android.content.Context;
import android.content.SharedPreferences;
import android.view.LayoutInflater;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import jp.chau2chaun2.kotlindatabindingsample.model.orma.OrmaDatabase;

@Module
public class AppModule {

    private final Context mContext;

    public AppModule(Context context) {
        mContext = context;
    }

    @Provides
    Context provideContext() {
        return mContext;
    }

    @Provides
    LayoutInflater provideLayoutInflater(Context context) {
        return LayoutInflater.from(context);
    }

    @Singleton
    @Provides
    SharedPreferences.Editor provideSharedPreference(Context context) {
        return context.getSharedPreferences("Default", Context.MODE_PRIVATE).edit();
    }
}

File: app/src/main/java/jp/chau2chaun2/kotlindatabindingsample/di/AppModule.kt
package jp.chau2chaun2.kotlindatabindingsample.di

import android.content.Context
import android.content.SharedPreferences
import android.view.LayoutInflater
import com.github.gfx.android.orma.AccessThreadConstraint
import dagger.Module
import dagger.Provides
import jp.chau2chaun2.kotlindatabindingsample.model.orma.OrmaDatabase
import javax.inject.Singleton

@Module
class AppModule(private val mContext: Context) {

    @Provides
    internal fun provideContext(): Context {
        return mContext
    }

    @Provides
    internal fun provideLayoutInflater(context: Context): LayoutInflater {
        return LayoutInflater.from(context)
    }

    @Singleton
    @Provides
    internal fun provideSharedPreference(context: Context): SharedPreferences.Editor {
        return context.getSharedPreferences("Default", Context.MODE_PRIVATE).edit()
    }

    @Singleton
    @Provides
    internal fun providesOrmaDatabase(context: Context): OrmaDatabase {
        return OrmaDatabase.builder(context)
                .writeOnMainThread(AccessThreadConstraint.NONE)
                .readOnMainThread(AccessThreadConstraint.NONE)
                .build()
    }
}
